### ISSUE TITLE
CSV file is valid as CSV

### ISSUE THREAD

**Issue by tombaker** [dcmi/dctap-python#2]
See https://github.com/dcmi/dctap/issues/28

The program should verify that the input file as valid as CSV:
- each row has the same number of columns

If errors found:
- stop program with message to user

**Comment by tombaker** [comment]
I started to try to solve this but scrapped my attempt when I found [goodtables](https://pypi.org/project/goodtables/). Features:
- Structural checks: Ensure that there are no empty rows, no blank headers, etc.
- Content checks: Ensure that the values have the correct types ("string", "number", "date", etc.), that their format is valid ("string must be an e-mail"), and that they respect the constraints ("age must be a number greater than 18").
- Support for multiple tabular formats: CSV, Excel files, LibreOffice, Data Package, etc.
- Parallelized validations for multi-table datasets
- Command line interface

Anyone have experience with this?
